lightning/src/chain/channelmonitor.rs

```diff
@@ -2512,14 +2524,24 @@ impl<Signer: WriteableEcdsaChannelSigner> ChannelMonitorImpl<Signer> {
 			}
 		}
 		if let Some(txid) = self.current_counterparty_commitment_txid {
-			if let Some(commitment_number) = self.counterparty_commitment_txn_on_chain.get(&txid) {
-				claim_htlcs!(*commitment_number, txid);
+			if txid == confirmed_spend_txid {
+				if let Some(commitment_number) = self.counterparty_commitment_txn_on_chain.get(&txid) {
```

Comment:
Sounds `counterparty_commitment_txn_on_chain` and `FundingSpendConfirmation` could be merged in a single data struct, as there is a redundancy we might store the same txid both as a `FundingSpendConfirmation { commitment_tx_to_counterparty_output` and hashmap entry.